sycl/include/sycl/ext/intel/esimd/detail/operators.hpp

```diff
@@ -482,6 +482,64 @@ __ESIMD_DEF_SIMD_VIEW_CMP_OP(>=, __ESIMD_DNS::is_simd_type_v<SimdT1>)
 
 #undef __ESIMD_DEF_SIMD_VIEW_CMP_OP
 
+////////////////////////////////////////////////////////////////////////////////
+// simd_mask global operators
+////////////////////////////////////////////////////////////////////////////////
+
+// ========= simd_mask arithmetic operators
+
+#define __ESIMD_DEF_SIMD_MASK_BIN_OP(BINOP, COND)                              \
+  /* simd* BINOP simd_mask<*...> */                                            \
+  template <class SimdT1, class SimdT2,                                        \
+            class T1 = typename SimdT1::element_type,                          \
+            class T2 = typename SimdT2::element_type,                          \
+            class = std::enable_if_t<                                          \
+                __ESIMD_DNS::is_simd_obj_impl_derivative_v<SimdT1> &&          \
+                __ESIMD_DNS::is_simd_mask_type_v<SimdT2> &&                    \
+                (SimdT1::length == SimdT2::length) && COND>>                   \
+  inline SimdT1 operator BINOP(const SimdT1 &LHS, const SimdT2 &RHS) {         \
```

Comment:
simd<float,N> a;
simd_mask<N> m;
auto res = a * m;
Now it is lowered to:
simd<float, N> m_float = m;
res = a * m_float;
The alternative sequence would be:
res = a.merge(simd<float,N>(0))

We need to choose the lowering depending on which one creates more efficient final code. Can you please do some comparison here?

Reply:
Merge version is better as far as it can be baled with another instructions.